server/container/guice/cassandra-guice/src/main/java/org/apache/james/modules/mailbox/ElasticSearchStartUpCheck.java

```diff
@@ -37,23 +37,41 @@
     private static final Logger LOGGER = LoggerFactory.getLogger(ElasticSearchStartUpCheck.class);
     private static final String CHECK_NAME = "ElasticSearchStartUpCheck";
     private static final String CAN_NOT_FIND_VERSION_MESSAGE = "cannot find ElasticSearch version";
+    private static final String SUGGESTION_VERSION_IS_NOT_MATCHED_MESSAGE = "suggested ElasticSearch version (%s) is not match " +
+        "with current version (%s)";
+
+    private static final String ES_VERSION_SUGGESTION ="2.4.6";
 
     private final Host esHttpHost;
+    private final String esVersionSuggestion;
 
     @Inject
     public ElasticSearchStartUpCheck(ESReporterConfiguration esConfiguration) {
-        this.esHttpHost =  Host.parseConfString(esConfiguration.getHostWithPort());
+        this(Host.parseConfString(esConfiguration.getHostWithPort()), ES_VERSION_SUGGESTION);
     }
 
     @VisibleForTesting
-    ElasticSearchStartUpCheck(Host esHttpHost) {
+    ElasticSearchStartUpCheck(Host esHttpHost, String esVersionSuggestion) {
         this.esHttpHost = esHttpHost;
+        this.esVersionSuggestion = esVersionSuggestion;
     }
 
+
     @Override
     public CheckResult check() {
         try {
             About.Version version = getVersion();
+            if (!version.getNumber().equals(ES_VERSION_SUGGESTION)) {
```

Comment:
Mmmh I am not sure we should be too restrictive about this...

Actually we accept 2.4.x so starting with a 2.4.7 should be OK. Maybe even from a 2.5.2 but 3.x should be rejected.

I believe we need to parse version number into business objects which we can support advanced comparaisons on...